src/geofabrics/processor.py

```diff
@@ -122,40 +146,83 @@ def get_vector_paths(self, key) -> list:
                 paths.append(self.instructions['instructions']['data_paths'][key])
 
         # Check the instructions for LINZ hoster vector data
-        if self.check_apis("linz") and key in self.instructions['instructions']['apis']['linz']:
-
-            assert self.check_instruction_path('local_cache'), "Local cache file path must exist to specify the " + \
-                "location to download vector data from the LINZ API"
-            assert self.catchment_geometry is not None, "The `self.catchment_directory` object must exist before a" + \
-                "vector is downloaded using `vector.Linz`"
-
-            # Key included the LINZ APIs - download data then add
-            vector_instruction = self.instructions['instructions']['apis']['linz'][key]
-            vector_fetcher = geoapis.vector.Linz(self.instructions['instructions']['apis']['linz']['key'],
-                                                 self.catchment_geometry.catchment, verbose=True)
-            cache_dir = pathlib.Path(self.get_instruction_path('local_cache'))
-            geometry_type = vector_instruction['type']
-
-            # Cycle through all layers specified - save each and add to the path list
-            for layer in vector_instruction['layers']:
-                vector = vector_fetcher.run(layer, geometry_type)
-
-                # Ensure directory for layer and save vector file
-                layer_dir = cache_dir / str(layer)
-                layer_dir.mkdir(parents=True, exist_ok=True)
-                vector_dir = layer_dir / key
-                vector.to_file(vector_dir)
-                shutil.make_archive(base_name=vector_dir, format='zip', root_dir=vector_dir)
-                shutil.rmtree(vector_dir)
-                paths.append(layer_dir / f"{key}.zip")
+        data_services = {"linz": geoapis.vector.Linz, "lris": geoapis.vector.Lris}  # API name and geoapis class pairs
+        for data_service in data_services.keys():
+            if self.check_apis(data_service) and key in self.instructions['instructions']['apis'][data_service]:
+                cache_dir = pathlib.Path(self.get_instruction_path('local_cache'))
+                api_key = self.instructions['instructions']['apis'][data_service]['key']
+
+                assert self.check_instruction_path('local_cache'), "Local cache file path must exist to specify the" + \
+                    f" location to download vector data from the vector APIs: {data_services}"
+                assert self.catchment_geometry is not None, "The `self.catchment_directory` object must exist " + \
+                    "before avector is downloaded using `vector.Linz`"
```

Comment:
Add some comment to explain your handling of apis keys here

Reply:
Done - in next commit :)